File: DSA/Raima_Rekhi_590013023/Day1Question2.c
#include <stdio.h>

void find_Missing_And_Repeating(int arr[],int n);
int main(){
int n;
printf("Enter the size of the array:\n");
scanf("%d",&n);
int arr1[n];

// Prompt the user to enter the elements of the array
printf("Enter the elements of the array: \n");
for(int i=0;i<n;++i)
{
 scanf("%d",&arr1[i]);
}
find_Missing_And_Repeating(arr1,n);
}

// Function which finds and prints the missing and repeating numbers in the array using frequency counting.
void find_Missing_And_Repeating(int arr[],int n){
int count[n+1]; // count[0] will be unused
for(int i=0;i<=n;++i){
count[i]=0;
}
// Count the frequency of each number
for(int i=0;i<n;++i){
count[arr[i]]++;
}

int missing=-1,repeating=-1;
for(int i=1;i<=n;++i){
if(count[i]==0){
missing=i;
}
else if(count[i]==2){
repeating=i;
}
}

printf("Number repeating twice: %d\n", repeating);
printf("Missing number: %d\n", missing);
}

File: DSA/AnshiAgrawal_500124498/Day_1/Day1Question2.java
public class Day1Question2 {
    public static void main(String[] args) {
        int[] universal = {1, 2, 3, 4, 5};
        int[] arr = {1, 3,2 , 2, 3};
        int i,j;
        for(i=0;i<universal.length;i++){
            int count=0;
            for (j=0;j<arr.length;j++){
                if(universal[i]== arr[j]) {
                    count++;
                }
            }
            if(count==0)
                System.out.println("missing:"+universal[i]);
            else if(count==2)
                System.out.println("Twice:"+universal[i]);

        }
    }
}
